invenio_app/ext.py

```diff
@@ -15,13 +15,17 @@
 import pkg_resources
 from flask import Blueprint, g, request
 from flask_limiter import Limiter
-from flask_limiter.util import get_ipaddr
 from flask_talisman import Talisman
 from werkzeug.utils import import_string
 
 from . import config
 
 
+def limit_with_ua_and_ip():
+    """Create key for the rate limiting."""
+    return str(request.user_agent) + request.remote_addr
```

Comment:
@inveniosoftware/architects In combination with the `RATELIMIT_APPLICATION` config above, this applies an application-wide rate-limit, which combined with e.g. a Redis storage backend would get us the desired 5000 reqs/hour/user.

Since this is meant to be a sensible default to be applied throughout all Invenio instances, we should probably better define what is a good value and what should a "user" count as. In this case, IP + User-Agent is a pretty common case which e.g. wouldn't block a university campus (i.e. one IP but different User-Agents) from accessing a resource.

Reply:
I am not sure 5K per hour is enough, in one of hour instance I have count about 200 requests just after visiting 4 web pages, basically 4 clicks.

Reply:
If most of these 200 requests are `/static/*` you're ok, since in production they get served by nginx, and for local develeopment I believe `Flask-Limiter` excludes them.

Reply:
You are totally right in normal conditions, the only issue we have is that the deposit interface, if kept open, performs some pulling every `x` seconds.
I think we can go with this and eventually we will adjust accordingly.